container_opts.go

```diff
@@ -127,3 +127,21 @@ func setSnapshotterIfEmpty(c *containers.Container) {
 		c.Snapshotter = DefaultSnapshotter
 	}
 }
+
+// WithContainerExtension appends extension data to the container object.
+// Use this to decorate the container object with additional data for the client
+// integration.
+//
+// Make sure to register the type of `extension` in the typeurl package via
```

Comment:
Sorry, I should have been more clear :sweat_smile: 

If the url was not registered via `typeurl.Register` and the message doesn't implement `protobuf/proto.Message` this will error out :)

Reply:
We should probably be able to support json here, as well.